great_expectations/datasource/fluent/snowflake_datasource.py

```diff
@@ -204,6 +210,17 @@ def get_execution_engine(self) -> SqlAlchemyExecutionEngine:
 
     @override
     def get_engine(self) -> sqlalchemy.Engine:
+        # This import is here to avoid a circular import
+        from great_expectations.data_context import CloudDataContext
+
+        # This is used to set the application query parameter, which provides attribution to GX
+        # for the Snowflake Partner program.
+        if isinstance(self._data_context, CloudDataContext):
+            # TODO: Change this value to ""great_expectations_cloud" once configured in Snowflake partner account
+            snowflake_partner_application = "great_expectations_oss"
+        else:
+            snowflake_partner_application = "great_expectations_oss"
```

Comment:
Can we make these constants?

Reply:
And is there a way to decouple Cloud logic here? I don't love leaking this logic here but we might need to given our current architecture

Reply:
Sure. Added constants in https://github.com/great-expectations/great_expectations/pull/9863/commits/3f3dd3c4885bec2d3f2b8b55b4619129dc66b346. 

WRT second point, I don't think there's a cleaner way to decouple cloud logic, but I did factor it out into a separate util method so at least it's less exposed in the public method. https://github.com/great-expectations/great_expectations/pull/9863/commits/10066e7ba7b7b543a32bd7dfcb83a2471dc0242f